common/lib/capa/capa/capa_problem.py

```diff
@@ -855,17 +870,76 @@ def _preprocess_problem(self, tree):  # private
                 id=response_id_str
             )
 
-            # assign one answer_id for each input type or solution type
+            # assign one answer_id for each input type
             for entry in inputfields:
                 entry.attrib['response_id'] = str(response_id)
                 entry.attrib['answer_id'] = str(answer_id)
                 entry.attrib['id'] = "%s_%i_%i" % (self.problem_id, response_id, answer_id)
                 answer_id = answer_id + 1
 
-            # Find the label and save it for html transformation step
-            responsetype_label = response.find('label')
-            problem_data[self.problem_id + '_' + str(response_id)] = {
-                'label': responsetype_label.text if responsetype_label is not None else ''
+            question_id = self.problem_id + '_' + str(response_id)
+            label = ''
+            element_to_be_deleted = None
+
+            # Extract label value from <label> tag or label attribute from inside the responsetype
+            responsetype_label_tag = response.find('label')
+            if responsetype_label_tag is not None:
+                label = responsetype_label_tag.text
```

Comment:
Looks like the value of label would be question title, can we rename `label` to `question_title` or something similar ?